### PR TITLE
Gherkin 7

### PR DESCRIPTION
**Pull request by aslakhellesoy**
Upgrade to gherkin 7

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,15 +15,22 @@ Please visit [cucumber/CONTRIBUTING.md](https://github.com/cucumber/cucumber/blo
 ### Added
 
 ### Changed
-* Going forward the minimum ruby version for all cucumber based gems is 2.3
-([luke-hill](https://github.com/luke-hill))
+* Going forward the minimum ruby version for all cucumber based gems is 2.4
```
Comment (luke-hill) [inline CHANGELOG.md]: Is this strictly true for all cucumber gems (Are we going to make the minimum supported ver 2.4?

I don't mind either way, but I didn't feel comfortable making that decision as I know a reasonable amount of people use 2.3 and I'm still new-ish here.

I'm definitely all for it, as it makes support 10* easier. But just wanted to confirm this was for *all* cucumber gems, not just the main one.

File: cucumber.gemspec [hunk 2]
```diff
@@ -18,34 +18,34 @@ Gem::Specification.new do |s|
...
 
   # For maintainer scripts
-  s.add_development_dependency 'octokit'
+  s.add_development_dependency 'octokit', '~> 4.14', '>= 4.14.0'
 
   # Needed for examples (rake examples)
-  s.add_development_dependency 'capybara', '>= 2.1'
-  s.add_development_dependency 'rack-test', '>= 0.6.1'
-  s.add_development_dependency 'sinatra', '>= 1.3.2'
+  s.add_development_dependency 'capybara', '~> 3.28', '>= 3.28.0'
```
Comment (luke-hill) [inline cucumber.gemspec]: this is a sizeable bump. Are we sure we're happy bumping everything immediately to their latest versions

Comment (aslakhellesoy) [reply cucumber.gemspec]: Yes we are, as long as it doesn't break the build.

It looks like none of the `/examples` run in CI. Nor can I run them locally. I'll see if I can make them work. If not we might as well remove the capybara dependency

File: CHANGELOG.md (hunk 1 above)
Comment (aslakhellesoy) [reply CHANGELOG.md]: I reverted to 2.3

**Pull request merged**